geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-api/src/main/scala/org/locationtech/geomesa/fs/storage/api/package.scala

```diff
@@ -65,6 +87,23 @@ package object api {
       * @param partition partition to compact, or all partitions
       * @param threads suggested threads to use for file system operations
       */
-    def compact(partition: Option[String], threads: Int = 1)
+    @deprecated("replaced with compact(Option[String], Option[Long], Int)")
+    def compact(partition: Option[String], threads: Int): Unit
```

Comment:
Should the signature remain 
```def compact(partition: Option[String], threads: Int = 1)```?

If it does not, wouldn't this break folks calling out with like
`compact(partition)`?

Reply:
I think scala won't let you have default values for overloaded functions. But since the new method has defaults for both `threads` and `fileSize` that invocation will still work. I think all of the following will still work:
`compact(partition)`, `compact(partition, 1)`, `compact(partition, threads = 1)`

Reply:
Ah!  I missed that the new function has two default parameters (which Scala likely desugars to 3 different functions).  Makes sense!